src/commands/commit/promote.ts

```diff
@@ -1,27 +1,82 @@
 import { Flags } from "@oclif/core";
 
 import * as ApiV1 from "@/lib/api-v1";
-import BaseCommand from "@/lib/base-command";
-import { withSpinner } from "@/lib/helpers/request";
+import BaseCommand, { TFlags } from "@/lib/base-command";
+import { ApiError } from "@/lib/helpers/error";
+import {
+  formatErrorRespMessage,
+  isSuccessResp,
+  withSpinner,
+} from "@/lib/helpers/request";
+import { indentString } from "@/lib/helpers/string";
 import { promptToConfirm } from "@/lib/helpers/ux";
 
 export default class CommitPromote extends BaseCommand<typeof CommitPromote> {
-  static summary = "Promote all changes to the destination environment.";
+  static summary =
+    "Promote a single commit to the subsequent environment or all changes to the destination environment.";
 
   static flags = {
     to: Flags.string({
       summary:
-        "The destination environment to promote changes from the preceding environment.",
-      required: true,
+        "The destination environment to promote all changes from the preceding environment.",
     }),
     force: Flags.boolean({
       summary: "Remove the confirmation prompt.",
     }),
+    only: Flags.string({
+      summary: "The ID of the commit to promote to the subsequent environment",
+    }),
   };
 
   async run(): Promise<void> {
     const { flags } = this.props;
 
+    // Currently this command support two modes of operation based on the flags:
+    // * --to : Promotes all changes to the destination environment.
+    // * --only: Promotes one commit to the subsequent enviroment
+    // The absence or presence of both flags will result in an error.
+    if (flags.to && flags.only) {
+      throw new Error(
+        "`--to` and `--only` flags cannot be used together.\n See more help with --help",
+      );
+    }
+
+    if (!flags.to && !flags.only) {
+      throw new Error(
+        "You must specify either `--to` or `--only` flag.\n See more help with --help",
+      );
+    }
+
+    return flags.only
+      ? this.promoteOneChange(flags)
+      : this.promoteAllChanges(flags);
```

Comment:
```suggestion
      ? this.promoteOneChange()
      : this.promoteAllChanges();
```
Nit, you don't need to pass in flags. You can access them via `this.props.flags`.

Reply:
Gotcha!